lib/ansible/module_utils/facts/system/distribution.py

```diff
@@ -310,9 +310,24 @@ def parse_distribution_file_SUSE(self, name, data, path, collected_facts):
                         suse_facts['distribution_release'] = release.group(1)
                         suse_facts['distribution_version'] = collected_facts['distribution_version'] + '.' + release.group(1)
 
-        # See https://www.suse.com/support/kb/doc/?id=000019341 for SLES for SAP
-        if os.path.islink('/etc/products.d/baseproduct') and os.path.realpath('/etc/products.d/baseproduct').endswith('SLES_SAP.prod'):
-            suse_facts['distribution'] = 'SLES_SAP'
+        # Check VARIANT_ID first for SLES4SAP or SL-Micro
+        variant_id_match = re.search(r'^VARIANT_ID="?([^"\n]*)"?', data, re.MULTILINE)
+        if variant_id_match:
+            variant_id = variant_id_match.group(1)
+            if variant_id == 'server-sap':
+                suse_facts['distribution'] = 'SLES_SAP'
+            elif variant_id == 'sles-sap':
+                suse_facts['distribution'] = 'SLES_SAP'
```

Comment:
How about adding a tuple of expected outputs - 

```suggestion
            if variant_id in ('server-sap', 'sles-sap'):
                suse_facts['distribution'] = 'SLES_SAP'
```